bitnami/redis/templates/configmap.yaml

```diff
@@ -51,3 +51,69 @@ data:
 {{- tpl .Values.sentinel.configmap . | nindent 4 }}
 {{- end }}
 {{- end }}
+---
+{{- if .Values.sentinel.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "redis.fullname" . }}-shutdown-script
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "redis.name" . }}
+    chart: {{ template "redis.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+data:
+  shutdown.sh: |-
+    #!/bin/bash
+    MASTER_SET="{{ .Values.sentinel.masterSet }}"
+    REDIS_COMMAND="redis-cli -p 26379"
+
+    function get_master_ip {
+      echo $($REDIS_COMMAND sentinel get-master-addr-by-name mymaster | head -n 1)
+    }
+
+    function check_master_election {
+      MY_IP=$1
+      MASTER_SET=$2
+
+      if [ $(get_master_ip $MASTER_SET) == $MY_IP ]; then
+        echo "This instance is current master"
+      fi
+
+      if [ $(get_master_ip $MASTER_SET) == "127.0.0.1" ]; then
+        echo "Current master is set to 127.0.0.1"
+      fi
+
+      if [ "$(timeout 1 redis-cli -h $(get_master_ip $MASTER_SET) -p 6379 -a $(cat /opt/bitnami/redis/secrets/redis-password) PING)" != "PONG" ]; then
```

Comment:
Keeping in mind the style improvements mentioned throughout the review of the .sh file, is it guaranteed that `/opt/bitnami/redis/secrets/redis-password` will exist?